lib/src/Base/Stat/SampleImplementation.cxx

```diff
@@ -1883,80 +1810,43 @@ Scalar SampleImplementation::computeEmpiricalCDF(const Point & point,
   return static_cast < Scalar > (functor.accumulator_) / size_;
 }
 
-struct MaxPerComponentPolicy
-{
-  typedef Point value_type;
-
-  static inline value_type GetInvariant(const SampleImplementation & nsi)
-  {
-    return value_type(nsi.getDimension(), - std::numeric_limits<Scalar>::max());
-  }
-
-  template <typename T>
-  static inline value_type & inplace_op( value_type & a, const T & b)
-  {
-    const UnsignedInteger dim = a.getDimension();
-    for (UnsignedInteger j = 0; j < dim; ++j) a[j] = std::max( a[j], b[j] );
-    return a;
-  }
-}; /* end struct MaxPerComponentPolicy */
-
-struct MinPerComponentPolicy
-{
-  typedef Point value_type;
-
-  static inline value_type GetInvariant(const SampleImplementation & nsi)
-  {
-    return value_type(nsi.getDimension(), std::numeric_limits<Scalar>::max());
-  }
-
-  template <typename T>
-  static inline value_type & inplace_op( value_type & a, const T & b)
-  {
-    const UnsignedInteger dim = a.getDimension();
-    for (UnsignedInteger j = 0; j < dim; ++j) a[j] = std::min( a[j], b[j] );
-    return a;
-  }
-}; /* end struct MinPerComponentPolicy */
-
 
 /* Maximum accessor */
 Point SampleImplementation::getMax() const
 {
   if (size_ == 0) throw InternalException(HERE) << "Impossible to get the maximum of an empty Sample";
 
-  ReductionFunctor<MaxPerComponentPolicy> functor( *this, MaxPerComponentPolicy() );
-  functor.accumulator_ = operator[](0);
-  TBB::ParallelReduce( 1, size_, functor );
-  return functor.accumulator_;
+  Point maxPoint(dimension_, - std::numeric_limits<Scalar>::max());
```

Comment:
@dbarbier In all other places we use -SpecFunc::MaxNumericalScalar. What is the rational of not using it here?

Reply:
Copy/paste from MaxPerComponentPolicy